zcash_primitives/src/transaction/builder.rs

```diff
@@ -1017,6 +1021,23 @@ where
         .map_err(Error::OrchardBuild)
 }
 
+#[cfg(zcash_unstable = "nu6" /* TODO nu7 */ )]
+fn first_nullifier(
+    orchard_bundle: &Option<OrchardBundle<Authorized>>,
+) -> Result<&Nullifier, io::Error> {
+    match orchard_bundle {
+        Some(OrchardBundle::OrchardVanilla(_)) => Err(io::Error::new(
+            io::ErrorKind::Other,
+            "Incorrect bundle type",
```

Comment:
the function is named `first_nullifier`, why block it for Vanilla?

Reply:
It is only relevant for ZSA/Swap, in case of Vanilla you never call this method and if you managed somehow - it will just panic

Reply:
The flow is relevant only to ZSA, but  in the local scope, it makes more sense to return the first nullifier for both types instead of an undocumented error.
Add comment or return the first nullifier for Vanilla.